server/container/guice/jpa-common-guice/src/test/java/org/apache/james/TestJPAConfigurationModuleWithSqlValidation.java

```diff
@@ -19,32 +19,85 @@
 
 package org.apache.james;
 
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+
 import javax.inject.Singleton;
 
 import org.apache.james.modules.data.JPAConfiguration;
 
 import com.google.inject.AbstractModule;
 import com.google.inject.Provides;
 
-public class TestJPAConfigurationModuleWithSqlValidation extends AbstractModule {
+public interface TestJPAConfigurationModuleWithSqlValidation {
+
+    class NoDatabaseAuthentication extends AbstractModule {
+        @Override
+        protected void configure() {
+        }
+
+        @Provides
+        @Singleton
+        JPAConfiguration provideConfiguration() {
+            return jpaConfigurationBuilder().build();
+        }
+    }
+
+    class WithDatabaseAuthentication extends AbstractModule {
 
-    private static final String JDBC_EMBEDDED_URL = "jdbc:derby:memory:mailboxintegration;create=true";
-    private static final String JDBC_EMBEDDED_DRIVER = org.apache.derby.jdbc.EmbeddedDriver.class.getName();
-    private static final String VALIDATION_SQL_QUERY = "VALUES 1";
+        @Override
+        protected void configure() {
+            setupAuthenticationOnDerby();
+        }
 
-    @Override
-    protected void configure() {
+        @Provides
+        @Singleton
+        JPAConfiguration provideConfiguration() {
+            return jpaConfigurationBuilder()
+                    .username(DATABASE_USERNAME)
+                    .password(DATABASE_PASSWORD)
+                    .build();
+        }
+
+        private void setupAuthenticationOnDerby() {
+            try {
+                Connection conn = DriverManager.getConnection(JDBC_EMBEDDED_URL, DATABASE_USERNAME, DATABASE_PASSWORD);
+                Statement s = conn.createStatement();
+                // Setting and Confirming requireAuthentication
+                s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY('derby.connection.requireAuthentication', 'true')");
```

Comment:
what about having a function to avoid repeating `s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY` ?